query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs

```diff
@@ -0,0 +1,389 @@
+            serde_json::json!({
+                "data": {
+                "findFirstResource": {
+                  "occStamp": 0,
+                  "id": 1
+                }
+              }
+            })
+        } else {
+            serde_json::json!({
+                "data": {
+                "findFirstResource": {
+                  "occStamp": 1,
+                  "id": 1
+                }
+              }
+            })
+        };
+        assert_eq!(res, expected);
+
+        Ok(())
+    }
+
+    async fn update_and_delete(runner: Arc<Runner>) {
+        update_one_resource(runner.clone()).await;
+        delete_one_resource(runner).await;
+    }
+
+    async fn create_one_resource(runner: Arc<Runner>) {
+        let create_one_resource = r#"
+        mutation {
+            createOneResource(data: {id: 1}) {
+              id
+            }
+          }"#;
+
+        runner.query(create_one_resource).await.unwrap().to_json_value();
+    }
+
+    async fn update_one_resource(runner: Arc<Runner>) -> serde_json::Value {
+        let update_one_resource = r#"
+        mutation {
+            updateOneResource(data: {occStamp: {increment: 1}}, where: {occStamp: 0}) {
+              occStamp,
+              id
+            }
+          }
+        "#;
+
+        runner.query(update_one_resource).await.unwrap().to_json_value()
+    }
+
+    #[allow(dead_code)]
+    async fn upsert_one_resource(runner: Arc<Runner>) -> serde_json::Value {
+        let upsert_one_resource = r#"
+        mutation {
+            upsertOneResource(where: {occStamp: 0}, 
+             create: {
+               occStamp: 0,
+               id: 1
+             },
+             update: {
+                 occStamp: {increment: 1}
+               }) {
+             id,
+             occStamp
+            }
+           }
+        "#;
+
+        runner.query(upsert_one_resource).await.unwrap().to_json_value()
+    }
+
+    async fn delete_one_resource(runner: Arc<Runner>) -> serde_json::Value {
+        let delete_one_resource = r#"
+        mutation {
+            deleteOneResource(where: {occStamp: 0}) {
+              occStamp,
+              id
+            }
+          }
+        "#;
+
+        runner.query(delete_one_resource).await.unwrap().to_json_value()
+    }
+
+    async fn delete_many_resource(runner: Arc<Runner>) -> u64 {
+        let delete_many_resource = r#"
+        mutation {
+            deleteManyResource(where: {occStamp: 0}) {
+              count
+            }
+          }
+        "#;
+
+        let res = runner.query(delete_many_resource).await.unwrap().to_json_value();
+
+        res.as_object()
+            .unwrap()
+            .get("data")
+            .unwrap()
+            .as_object()
+            .unwrap()
+            .get("deleteManyResource")
+            .unwrap()
+            .as_object()
+            .unwrap()
+            .get("count")
+            .unwrap()
+            .as_u64()
+            .unwrap()
```

Comment:
We have a `run_query_json` macro that you could use here. It'd roughly look like so:

```rust
run_query_json!(&runner, "mutation { deleteManyResource() }", "data.deleteManyResource.count")
```

Either that, or, beware that `ops::Index` is implemented for `serde_json::Value`, meaning that you could refactor this whole thing with: `&res["data"]["deleteManyResource"]["count"].as_u64().unwrap()`.